### PR TITLE
[Lang] Support ti_print for wasm

### PR DESCRIPTION
**Pull request by squarefk**
Related issue = #

### CODE REVIEW

File: taichi/runtime/llvm/runtime.cpp [hunk 1]
```diff
@@ -1741,6 +1744,40 @@ i32 wasm_materialize(Context *context) {
...
+  ((i32 *)buffer)[print_pos * 2 + 1] = 0;  // 0 for i32
+  ((i32 *)buffer)[print_pos * 2 + 2] = value;
+}
+
+void wasm_print_f32(Context *context, f32 value) {
+  Ptr buffer = context->runtime->wasm_print_buffer;
+  if (buffer == nullptr)
+    return;
+  i32 total_cnt = ((i32 *)buffer)[0]++;
+  i32 print_pos = total_cnt % (1024 * 1024);
```
Comment (k-ye) [inline taichi/runtime/llvm/runtime.cpp:1766]: nit: name this to `kWasmPrintBufferSize`

File: taichi/runtime/llvm/runtime.cpp [hunk 2]
```diff
@@ -1741,6 +1744,40 @@ i32 wasm_materialize(Context *context) {
...
+  ((i32 *)buffer)[print_pos * 2 + 1] = 1;  // 1 for f32
+  ((f32 *)buffer)[print_pos * 2 + 2] = value;
+}
+
+void wasm_print_char(Context *context, i32 value) {
```
Comment (k-ye) [inline taichi/runtime/llvm/runtime.cpp:1771]: Can we pack four chars together to save some space?

File: taichi/runtime/llvm/runtime.cpp (hunk 1 above)
Comment (squarefk) [reply taichi/runtime/llvm/runtime.cpp]: https://github.com/taichi-dev/taichi/pull/2910/commits/147a2505cb388108b2c99833b12d95f04372b507

File: taichi/runtime/llvm/runtime.cpp (hunk 2 above)
Comment (squarefk) [reply taichi/runtime/llvm/runtime.cpp:1785]: The current struct works like
```
i32 total_print_character_num;
struct {
    int type;  // 0 for i32, 1 for f32, 2 for char
    union {
      i32 i;
      f32 f;
      char c;
    } data;
} wasm_buffer_buffer[kWasmPrintBufferSize];
```

Packed mode is hard to implement here... I added the above memory layout into the comment in taichi to help understand the design here. https://github.com/taichi-dev/taichi/pull/2910/commits/0413cfd4b19382c4fbd8c1e9bc215b20e08b38e8

Comment (k-ye) [reply taichi/runtime/llvm/runtime.cpp:1785]: It feels weird to take 8 bytes to store a single char... I think by padding the string to a multiple of 4, we can easily pack four chars? I.e.

```cpp
i32 total_print_character_num;
struct {
    int type;  // 0 for i32, 1 for f32, 2 for char
    union {
      i32 i;
      f32 f;
      i32 ch4;
    } data;
} wasm_buffer_buffer[kWasmPrintBufferSize];

```

**Review by k-ye — APPROVED** [review]
LGTM!

**Pull request merged**